### PR TITLE
Change the behavior of getSelfHostedSites, returns the sites accessed…

### PR DESCRIPTION
**Pull request by maxme**
… via XMLRPC, also changes the behavior of getWPComAndJetpackSites that now returns all sites accessed via REST API

### CODE REVIEW

**Comment by maxme** [comment]
`SiteStore.getVisibleWPComAndJetpackSites()` must be updated as well

File: example/src/test/java/org/wordpress/android/fluxc/site/SiteStoreUnitTest.java [hunk 1]
```diff
@@ -125,6 +125,52 @@ public void testHasSiteAndgetCountMethods() throws DuplicateSiteException {
...
+        jetpackOverRest.setSiteId(3);
+        jetpackOverRest.setIsWPCom(false);
+        jetpackOverRest.setIsJetpackInstalled(true);
+        jetpackOverRest.setIsJetpackConnected(true);
+        jetpackOverRest.setOrigin(SiteModel.ORIGIN_WPCOM_REST);
+        SiteSqlUtils.insertOrUpdateSite(jetpackOverRest);
+
+        assertEquals(3, mSiteStore.getSitesCount());
+        assertEquals(0, mSiteStore.getWPComSitesCount());
+        assertEquals(2, mSiteStore.getSelfHostedSitesCount());
```
Comment (aforcier) [inline example/src/test/java/org/wordpress/android/fluxc/site/SiteStoreUnitTest.java]: How about also adding a check for `mSiteStore.getWPComAndJetpackSitesCount()` here (and a few lines down in the same method)? For completeness' sake - once we drop `getJetpackSitesCount()`, we see `getSelfHostedSitesCount()` go down by 1 after we "install and connect" below, but nothing goes up.

File: example/src/test/java/org/wordpress/android/fluxc/site/SiteStoreUnitTest.java [hunk 2]
```diff
@@ -254,7 +282,7 @@ public void testGetSiteBySiteId() throws DuplicateSiteException {
         SiteSqlUtils.insertOrUpdateSite(wpComSite);
         SiteSqlUtils.insertOrUpdateSite(jetpackSite);
 
-        assertEquals(2, SiteSqlUtils.getWPComAndJetpackSites().getAsCursor().getCount());
+        assertEquals(1, SiteSqlUtils.getWPComAndJetpackSites().getAsCursor().getCount());
         assertNotNull(mSiteStore.getSiteBySiteId(wpComSite.getSiteId()));
         assertNotNull(mSiteStore.getSiteBySiteId(jetpackSite.getSiteId()));
```
Comment (aforcier) [inline example/src/test/java/org/wordpress/android/fluxc/site/SiteStoreUnitTest.java]: I think we should rename `jetpackSite` to `jetpackSiteOverXMLRPC` or something for clarity - and perhaps add a comment on this line that though not connected to a WP.com account in the app, we still expect a XML-RPC Jetpack site to tell us its (WP.com) site id.

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/SiteSqlUtils.java [hunk 3]
```diff
@@ -186,20 +183,31 @@ public ContentValues toCv(Boolean item) {
...
     public static SelectQuery<SiteModel> getSelfHostedSites() {
         return WellSql.select(SiteModel.class)
                 .where().beginGroup()
-                .equals(SiteModelTable.IS_WPCOM, false)
-                .equals(SiteModelTable.IS_JETPACK_CONNECTED, false)
+                .equals(SiteModelTable.ORIGIN, SiteModel.ORIGIN_XMLRPC)
                 .endGroup().endWhere();
     }
 
     public static SelectQuery<SiteModel> getWPComAndJetpackSites() {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/SiteSqlUtils.java]: Do you think we should rename `get.*WPComAndJetpackSites.*` methods too? They're more in line with `SiteModel.isUsingWPComRestApi` now. Maybe `get.*WPComRestApiSites.*`. I don't know 😕

File: example/src/test/java/org/wordpress/android/fluxc/site/SiteStoreUnitTest.java (hunk 1 above)
Comment (maxme) [inline example/src/test/java/org/wordpress/android/fluxc/site/SiteStoreUnitTest.java]: Yep and we'll rename `getWPComAndJetpackSitesCount` to `getSitesAccessedViaRESTCount` and see that one go up by 1 after "install and connect".

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/SiteSqlUtils.java (hunk 3 above)
Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/SiteSqlUtils.java]: Yes we'll rename them but not in this PR.

**Comment by aforcier** [comment]
Some of the connected tests in `ReleaseStack_SiteTestJetpack` need to be updated as well.

**Comment by aforcier** [comment]
Also in `ReleaseStack_SiteTestJetpack`, we should update [this comment](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/1939af38af3d07f620a0ab507cfee0c584a63bff/example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestJetpack.java#L273) - we do make a distinction now (it's right above an assertion that needs to be changed).

**Comment by aforcier** [comment]
Looks good! :shipit:

**Pull request merged**